### ISSUE TITLE
Parameters must be prefixed with - or --

### ISSUE THREAD

**Issue by andrewbaxter** [cbeust/jcommander#245]
This is not mentioned in the documentation and the error message isn't obvious.

Demonstration code:

``` java
class P1 {
    @Parameter(names = "hello")
    private int test;
}
P1 p1 = new P1();
JCommander j = new JCommander(p1);
j.parse(new String[]{"hello", "47"});
System.out.printf("command %s, test %d\n", j.getParsedCommand(), p1.test);
```

Raises:

```
Exception in thread "main" com.beust.jcommander.ParameterException: Was passed main parameter 'hello' but no main parameter was defined
    at com.beust.jcommander.JCommander.getMainParameter(JCommander.java:914)
    at com.beust.jcommander.JCommander.parseValues(JCommander.java:759)
    at com.beust.jcommander.JCommander.parse(JCommander.java:282)
    at com.beust.jcommander.JCommander.parse(JCommander.java:265)
    at test.main.main(main.java:18)
```

**Comment by jeremysolarz** [comment]
@cbeust what do you say? Shall this be added to the documentation, and if how? 😄

**Comment by cbeust** [comment]
Indeed, feel free to send a PR for the documentation.

Speaking of documentation, I'd also like to update the CSS so we can have the table of contents on the side and in a fixed position instead of at the top where it takes up most of the first screen before your scroll.

**Comment by jeremysolarz** [comment]
New docu looks really nice!

What do you think. Would put the "warning" that parameters have to be prefixed with - or -- directly in 1. Overview as a sub-point. Do you have a better idea?

**Comment by cbeust** [comment]
To be honest, I'm not sure it's worth mentioning, especially since it can
easily be improved to accept different characters if required.


-- 
Cédric

**Comment by jeremysolarz** [comment]
@cbeust ok then I leave it as is for now.

**Comment by cbeust** [comment]
Fixed in 1.61. Here is a version of your code above that works:

```
  static class P1 {
    @Parameter(names = "hello")
    private int test;
  }

  public static void main(String[] args) {
    P1 p1 = new P1();
    JCommander j = new JCommander();
    j.addCommand("p1", p1);
    j.parse(new String[]{"p1", "hello", "47"});
    System.out.printf("command %s, test %d\n", j.getParsedCommand(), p1.test);
}
```
